packages/create-block-tutorial-template/block-templates/render.php.mustache

```diff
@@ -1,10 +1,33 @@
-{{#isDynamicVariant}}
 <?php
 /**
+ * PHP file to use when rendering the block type on the server to show on the front end.
+ *
+ * The following variables are exposed to the file:
+ *     $attributes (array): The block attributes.
+ *     $content (string): The block default content.
+ *     $block (WP_Block): The block instance.
+ *
  * @see https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/block-api/block-metadata.md#render
  */
-?>
-<p <?php echo get_block_wrapper_attributes(); ?>>
-	<?php echo esc_html( $attributes['message'] ); ?>
-</p>
-{{/isDynamicVariant}}
+
+// Get the current year.
+$current_year = date( "Y" );
+
+// Determine which content to display.
+if ( isset( $attributes['fallbackCurrentYear'] ) && $attributes['fallbackCurrentYear'] === $current_year ) {
+
+	// The current year is the same as the fallback, so use the block content saved in the database (by the save.js function).
+	$block_content = $content;
+} else {
+
+	// The current year is different from the fallback, so render the updated block content.
+	if ( ! empty( $attributes['startingYear'] ) && ! empty( $attributes['showStartingYear'] ) ) {
+		$display_date = $attributes['startingYear'] . '–' . $current_year;
+	} else {
+		$display_date = $current_year;
+	}
+
+	$block_content = '<p ' . get_block_wrapper_attributes() . '>© ' . esc_html( $display_date ) . '</p>';
+}
+
+echo wp_kses_post( $block_content );
```

Comment:
It's always been a best practice to late escape anything that is being output to the browser that could be modified by a hook, filter, etc - https://developer.wordpress.org/apis/security/escaping/#toc_3

Reply:
If that would be an issue, then all core blocks would call `wp_kses_post`. However, it isn't the case. If this function is used, it's used to escape the content that comes from the user input when it is expected it might contain HTML. In the case of this block, it should be enough to cast the `startingYear` to the number and the rest should be safe.